### DISCUSSION TITLE
Error while calling a function in a Store

### DISCUSSION THREAD

**Discussion by busybox11** [alpinejs/alpine#3379] [1. Help] [answered]
Hello,
I have a store that is defined with the follwing code:
```JS
document.addEventListener('alpine:init', () => {
    Alpine.store('myStore', {
        IDs: [],

        add: (ID) => {
            if (!this.IDs.includes(ID)) {
                this.IDs.push(ID)
            }
        }
    })

    Alpine.data('app', () => ({
        // ...
    }))
})
```
As the code suggests, I also have an `x-data` declaration that is mandatory for the Store to work.
AlpineJS is deferred, and at the latest currently available version (`v3.10.5`).

I'm trying to call the `add` function from a function in one of my JS scripts, outside of Alpine, with the following call:
```JS
Alpine.store('myStore').add('test')
```
This is not working. I am getting the following error:
```
TypeError: Cannot read properties of undefined (reading 'includes')
    at Proxy.add (script.js:201:27)
    at script.js:438:56
```

Though, if I try to call the exact same function from my browser's devtools, the function works as expected, everything is reactive.

Am I missing something here?

**Comment by ekwoka** [comment, marked as answer]
Change the syntax to
```js
add(ID) {
  // function body
}
```

The arrow function prevents the `this` argument from referring to the parent object and would instead, in that context, refer to the `globalThis` otherwise called the `window`

**Reply by busybox11** [reply]
Ah, that makes sense. Thank you JavaScript for that

I probably could've figured this out by myself at some point lol.
Thank you for the help!
